public_html/wp-content/plugins/pattern-creator/src/plugins/save-post-modifier.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * WordPress dependencies
+ */
+import { dispatch, select } from '@wordpress/data';
+import { store } from '@wordpress/editor';
+import { useEffect, useState } from '@wordpress/element';
+import { registerPlugin } from '@wordpress/plugins';
+
+window.gutenbergSavePost = dispatch( store ).savePost;
+
+const SavePostModifier = () => {
+	const [ showModal, setShowModal ] = useState( false );
+
+	useEffect( () => {
+		// We don't want the publish sidebar confirmation
+		dispatch( store ).disablePublishSidebar();
+		/**
+		 * We currently don't have a hook to intercept Gutenberg save functionality
+		 * so we override the `savePost` function.
+		 *
+		 * This function is still called by gutenberg for autosaving so we need to handle that case as well.
+		 *
+		 * @param {Object} props
+		 * @param {boolean} props.isAutosave Whether autosave is calling this function.
+		 */
+		dispatch( store ).savePost = ( props ) => {
+			if ( props && props.isAutosave ) {
+				return;
+			}
+
+			// We need to wait until now to get the status otherwise we will be working with a previous context
+			const postStatus = select( store ).getEditedPostAttribute( 'status' );
+
+			// The post is set to 'publish' before 'savePost' is called,
+			// we can rely on that to make sure we only show the window when necessary
```

Comment:
I'm not sure I follow this logic - what sets the `postStatus` to publish?

Reply:
The Gutenberg `publish` button does. 

https://github.com/WordPress/gutenberg/blob/022d624665bb60e63a29cf235e788bf5d6a30845/packages/editor/src/components/post-publish-button/index.js#L138